lib/adapters/binance.py

```diff
@@ -0,0 +1,166 @@
+                    "outboundAccountPosition": self.position_update,
+                    "balanceUpdate": self.balance_update,
+                    "executionReport": self.order_report
+                }
+
+                message_map[payload["e"]](payload)
+            except KeyError:
+                msg = "unknown payload type {p}".format(p=payload.get("e"))
+                raise InvalidPayloadRecieved(msg)
+            except Exception as e:
+                raise Exception("unknown error occured in user stream", e)
+
+
+        def initial_rest_payload(self, payload) -> AccountStatus:
+            return AccountStatus(
+                CanWithdraw=payload["canWithdraw"],
+                CanTrade=payload["canTrade"],
+                CanDeposit=payload["canDeposit"],
+                Positions={
+                    balance["a"]: Position(
+                        Locked=float(balance["l"]),
+                        Free=float(balance["f"])
+                    ) for balance in payload["B"]
+                }
+            )
+
+
+        def account_update(self, payload) -> Tuple[str, AccountStatus]:
+            return UpdateType.ACCOUNT, AccountStatus(
+                CanWithdraw=payload["W"],
+                CanTrade=payload["T"],
+                CanDeposit=payload["D"],
+                Positions={
+                    balance["a"]: Position(
+                        Locked=float(balance["l"]),
+                        Free=float(balance["f"])
+                    ) for balance in payload["B"]
+                }
+            )
+
+
+        def balance_update(self, payload) -> Tuple[str, BalanceUpdate]:
+            return UpdateType.BALANCE, BalanceUpdate(
+                Asset=payload["a"], Update=float(payload["d"])
+            )
+
+
+        def position_update(self, payload) -> Tuple[str, PositionStatus]:
+            return UpdateType.POSITION, {
+                balance["a"]: Position(
+                    Locked=float(balance["l"]),
+                    Free=float(balance["f"])
+                ) for balance in payload["B"]
+            }
+
+
+        def order_report(self, payload):
+            # @todo - never did work out how to handle these.
+            pass
+
+    def connect(self, creds: Credentials, account: Account):
+        self.account = account
+        self.client = Client(creds['key'], creds['secret'])
+        self.socket_manager = BinanceSocketManager(self.client)
+
+        self.user_transformer = BinanceUserPayloadTransformer()
+        self.market_transformer = BinanceMarketStreamTransformer()
+        update = self.transformer.initial_rest_payload(
+            self.client.get_account()
+        )
+        account(UpdateType.ACCOUNT, update)
+
+        self.socket_manager.start_user_socket(self.handle_user_stream)
+
+
+    def run(self, strategy: Strategy, symbol: str):
+        self.strategy = strategy
+        self.socket_manager.start_symbol_ticker_socket(
```

Comment:
To think about later potentially: what if I have multiple symbols and wish to multiplex them?